src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,113 @@
+package duke;
+
+import duke.tasks.*;
+
+public class Parser {
+
+    /**
+     * Parses the command the user enters and execute it
+     * @param tasks TaskList that stores all the current tasks
+     * @param ui UI for interaction with the user
+     * @return boolean. true indicates that the user has terminated the program (by typing "bye")
+     * @throws DukeException for invalid user commands
+     */
+    public static boolean parse(TaskList tasks, Ui ui) throws DukeException {
+
+        String[] cmdString = ui.getUserCmd()
```

Comment:
Might be more readable to leave it in one line as the line is not very long yet.